pkg/actions/lua/databricks/client.go

```diff
@@ -141,24 +143,42 @@ func (client *Client) RegisterExternalTable(l *lua.State) int {
 	if metadata != nil {
 		metadataMap = metadata.(map[string]any)
 	}
-	status, err := client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location, metadataMap)
-	if err != nil {
-		if alreadyExists(err) {
-			err = client.deleteTable(catalogName, schemaName, tableName)
-			if err != nil {
-				lua.Errorf(l, "%s", err.Error())
-				panic("unreachable")
-			}
-			status, err = client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location, metadataMap)
-			if err != nil {
-				lua.Errorf(l, "%s", err.Error())
-				panic("unreachable")
-			}
-		} else {
+
+	// delete the table first if exists, to prevent "table already exists" errors in databricks
+	_, err := client.workspaceClient.Tables.GetByFullName(client.ctx, tableFullName(catalogName, schemaName, tableName))
+	if err != nil && !errors.Is(err, databricks.ErrResourceDoesNotExist) {
+		lua.Errorf(l, "%s", err.Error())
+		panic("unreachable")
+	}
+	if err == nil {
```

Comment:
comment on the condition at the end that if no error, table exists and we need to delete it before creation.

note: the previous version had it in a function that just was a call, you can keep the function if you like to include the "is exists" logic, but you need to report the error.